In [1]:
import pandas as pd
path = "GlobalWeatherRepository.csv"
df = pd.read_csv(path)

df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


In [2]:
df[
    [
        'temperature_fahrenheit',
        'pressure_mb',
        'air_quality_Sulphur_dioxide'
    ]
].corr()

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
temperature_fahrenheit,1.000000,-0.486833,-0.023152
pressure_mb,-0.486833,1.000000,0.018620
air_quality_Sulphur_dioxide,-0.023152,0.018620,1.000000


### Tugas 1 Analisis Korelasi

Berdasarkan hasil analisis korelasi, diperoleh:

- Temperature Fahrenheit dengan Pressure mb memiliki korelasi sebesar **-0.486833**, yang menunjukkan hubungan negatif sedang. Artinya, ketika temperature_fahrenheit meningkat, pressure_mb cenderung menurun.
- Temperature Fahrenheit dengan Air Quality Sulphur Dioxide memiliki korelasi sebesar **-0.023152**, yang menunjukkan hubungan negatif yang sangat lemah. Artinya, hampir tidak terdapat hubungan linear antara kedua variabel tersebut.
- Pressure mb dengan Air Quality Sulphur Dioxide memiliki korelasi sebesar **0.018620**, yang menunjukkan hubungan positif yang sangat lemah. Artinya, hampir tidak terdapat hubungan linear antara kedua variabel tersebut.

Dapat disimpulkan bahwa hubungan yang paling kuat di antara ketiga variabel tersebut adalah hubungan antara temperature_fahrenheit dan pressure_mb dengan nilai korelasi **-0.486833**.

In [5]:
import pandas as pd

path_customer = "shopping_trends.csv"
df_customer = pd.read_csv(path_customer)

df_customer

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,Cash,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,PayPal,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Credit Card,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,PayPal,Express,No,No,24,Venmo,Weekly


In [6]:
previous_purchases = df_customer['Previous Purchases']

print("Mean :", previous_purchases.mean())
print("Std  :", previous_purchases.std())
print("Min  :", previous_purchases.min())
print("Q1   :", previous_purchases.quantile(0.25))
print("Q2   :", previous_purchases.quantile(0.50))
print("Q3   :", previous_purchases.quantile(0.75))
print("Max  :", previous_purchases.max())

Mean : 25.35153846153846
Std  : 14.447125170462309
Min  : 1
Q1   : 13.0
Q2   : 25.0
Q3   : 38.0
Max  : 50


### Tugas 2a Analisis Previous Purchases

Berdasarkan hasil analisis statistik deskriptif pada data **Previous Purchases**, diperoleh:

- **Mean** = 25.351538
- **Std** = 14.447125
- **Min** = 1
- **Q1** = 13
- **Q2 (Median)** = 25
- **Q3** = 38
- **Max** = 50

Nilai mean sebesar **25.351538** menunjukkan bahwa rata-rata jumlah pembelian sebelumnya adalah sekitar 25 pembelian. Nilai standar deviasi sebesar **14.447125** menunjukkan adanya penyebaran data yang cukup besar dari nilai rata-ratanya.

Nilai minimum adalah **1** dan maksimum adalah **50**. Berdasarkan nilai kuartil, 25% data memiliki Previous Purchases kurang dari atau sama dengan **13**, 50% data kurang dari atau sama dengan **25**, dan 75% data kurang dari atau sama dengan **38**.

In [7]:
df_customer[
    ['Age', 'Purchase Amount (USD)']
].corr()

,Age,Purchase Amount (USD)
Age,1.000000,-0.010424
Purchase Amount (USD),-0.010424,1.000000


### Tugas 2b Korelasi Age vs Purchase Amount (USD)

Berdasarkan hasil analisis, korelasi antara **Age** dan **Purchase Amount (USD)** adalah sebesar **-0.010424**.

Nilai korelasi yang mendekati 0 menunjukkan bahwa hubungan linear antara usia pelanggan dan jumlah pembelian sangat lemah. Nilai negatif menunjukkan bahwa hubungan tersebut berarah negatif, tetapi hubungannya hampir tidak signifikan secara linear.

Dengan demikian, dapat disimpulkan bahwa **Age memiliki hubungan linear yang sangat lemah dengan Purchase Amount (USD)** pada dataset ini.

In [8]:
df_customer[
    ['Purchase Amount (USD)', 'Review Rating']
].corr()

,Purchase Amount (USD),Review Rating
Purchase Amount (USD),1.000000,0.030776
Review Rating,0.030776,1.000000


### Korelasi Tugas 2b Purchase Amount (USD) vs Review Rating

Berdasarkan hasil analisis, korelasi antara **Purchase Amount (USD)** dan **Review Rating** adalah sebesar **0.030776**.

Nilai korelasi yang mendekati 0 menunjukkan bahwa hubungan linear antara jumlah pembelian dan rating ulasan sangat lemah. Nilai positif menunjukkan hubungan searah, tetapi hubungan tersebut hampir tidak terdapat secara linear.

Dengan demikian, dapat disimpulkan bahwa **Purchase Amount (USD) memiliki hubungan linear yang sangat lemah dengan Review Rating** pada dataset ini.